utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -87,8 +87,12 @@ fun archive(inputDir: File, zipFile: File, prefix: String = ""): File =
 /**
  * Unpack the [File] to [targetDirectory] using [filter] to select only the entries of interest.
  */
-fun File.unpack(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) =
-    when (ArchiveType.getType(name)) {
+fun File.unpack(
+    targetDirectory: File,
+    fallbackArchiveType: ArchiveType = ArchiveType.NONE,
+    filter: (ArchiveEntry) -> Boolean = { true }
+) =
+    when (ArchiveType.getType(name).takeUnless { it == ArchiveType.NONE } ?: fallbackArchiveType) {
```

Comment:
I wonder whether we should really implement this as a fallback, or rather an override. An override would have the advantage that we could unpack wrong named archives. While that rarely happens, it seems like the more generic approach to always use the `ArchiveType` unless it is `NONE`. What do you think?

Reply:
Yes, this is better. Switched to override logic.